### PR TITLE
Dynamic storage

### PR DESCRIPTION
**Pull request by AlbyGiaco**

### CODE REVIEW

**Review by nicolas-racchi — APPROVED** [review]

**Review by RiccardoGalbusera — APPROVED** [review]

File: contracts/PositionManager.sol [hunk 1]
```diff
@@ -197,23 +197,19 @@ contract PositionManager is IPositionManager, ERC721Holder, Initializable {
...
-
-        return aaveUserReserves.tokenIds[id];
-    }
-
-    ///@notice returns array of positions moved to aave
-    ///@return Storage.AavePositions return the array of positions moved on aave
-    function getAavePositionsArray() external view override returns (AavePositions[] memory) {
-        StorageStruct storage Storage = PositionManagerStorage.getStorage();
-        return Storage.aavePositionsArray;
+    function getAaveDataFromTokenId(uint256 tokenId) external view override returns (uint256, address) {
```
Comment (luduvigo) [inline contracts/PositionManager.sol:200]: Why did we not put the natspec here?

File: contracts/actions/AaveDeposit.sol [hunk 2]
```diff
@@ -51,25 +49,45 @@ contract AaveDeposit is IAaveDeposit {
 
         shares = aToken.scaledBalanceOf(address(this)) - balanceBefore;
 
-        id = _updateAavePosition(token, shares, tokenId);
-        emit DepositedOnAave(address(this), token, id, shares);
+        _updateAavePosition(token, shares, tokenId);
+        emit DepositedOnAave(address(this), token, tokenId, shares);
     }
 
+    ///@dev now dont use anymore aaveid since we have tokenId that is unique - store shares for tokenId and tokenToAave for tokenId
```
Comment (luduvigo) [inline contracts/actions/AaveDeposit.sol:56]: if we write _**can't**_ let's write _**don't**_

File: contracts/utils/Storage.sol [hunk 3]
```diff
@@ -261,4 +260,56 @@ library PositionManagerStorage {
...
+        StorageStruct storage ds = getStorage();
+        bytes16 y;
+
+        assembly {
+            y := shl(128, hashedKey)
+        }
+
+        bytes32 storageVariableHash = ds.storageVars[uint128(y)];
+
+        require(storageVariableHash == hashedKey, 'SDK');
```
Comment (luduvigo) [inline contracts/utils/Storage.sol:276]: Can you explain me this require?

File: contracts/actions/AaveDeposit.sol (hunk 2 above)
Comment (RiccardoGalbusera) [reply contracts/actions/AaveDeposit.sol:56]: also, is it correct to use natspec to clarify updates in the code? Shouldn't it be readable for someone who looks at the code for the first time?

**Review by luduvigo — APPROVED** [review]

**Pull request merged**